designs/virtualization/4.x/create-vm-dropdowns/create-vm-dropdowns.md

```diff
@@ -0,0 +1,137 @@
+
+![Create VM - OS disk modal](img/1-3-1.jpg)
+Editing that disk will display an info notification within the edit modal, mentioning the link between this disk and the one created on the OS source step.
+The fields related to the OS source, in this case, the URL, will be locked.
+
+## Netwroking step
+
+![Create VM - Networking -list](img/2-0.jpg)
+NIC list.
+For any VM source selection (with the exception of PXE) a default NIC will be added.
+This NIC exposes the VM to the cluster.
+an info alert will notify the user of the purpose of this NIC and that modifying this NIC might break that 'exposing to cluster' configuration.
+
+![Create VM - Networking -list](img/2-5.jpg)
+On deleting the default NIC, a confirmation modal will present itsself.
+
+![Create VM - Networking -Default](img/2-1.jpg)
+Edit modal.
+The Advanced drawer is always collapsed by default
+
+![Create VM - Networking -Add](img/2-2.jpg
+The Add NIC modal.
+The Advanced drawer is closed by default, as seen in the previous image.
+All fields have moved into the advanced drawer, except the Name and Network fields.
+Network field does not require description for each option since these options are created and managed by the user.
+
+![Create VM - Networking -Model](img/2-3.jpg)
+'Model' dropdown options.
+ViritIO marked as (recommended)
+
+![Create VM - Networking -Binding](img/2-4.jpg)
+'Binding method' dropdown options (previously 'Type')
+
+This field's default is dependent on the Network field.
+For 'Pod network' the default option will be 'Masquerade'.
+For other Network option, the default will be 'Birgde'.
+
+## Storage
+
+![Create VM - Storage -list](img/3-0.jpg)
+Disks list.
+In case no disk type OS image source selected in the General step, a default disk will be created in order to make the VM operational.
+This will be highlighted by an in-line info alert (like in the Networking step)
+
+![Create VM - Storage -Defaul disk deletion](img/3-4.jpg)
+On deleting the default disk, a confirmation modal will present itsself.
+
+
+![Create VM - Storage -Default disk](img/3-0-0.jpg)
+
+Add disk modal.
+- 'Type' field changed to 'Disk content'
+- 'Name' field changed to 'Disk name'
+- 'Size' changed to 'Disk size (PVC size)'
+- Some fields pulled into the the Advanced drawer (see image below)
+
+![Create VM - Storage -list](img/3-1.jpg)
+The Add/Edit modal expanded view.
+- Advanced drawer added containing: 
+    - Interface
+    - Volume mode
+    - Access mode (should be automatically selected and locked to the Storage class default)
+    - Make disk bootable checkbox (new)
+
+![Create VM - Storage -list](img/3-2.jpg)
+Disk content dropdwon options.
+
+![Create VM - Storage -list empty state](img/3-3.jpg)
+List empty state will notify the user that at least one disk is needed.
+This will be supported by an alert blocking the user from moving from this step when there are no disks.
+
+## Review & create
+
+ ![Create VM - Advanced -drawers](img/5-0.jpg)
```

Comment:
I've been talking to @jelkosz about the "create a new template from this configuration" 
He made a good point about not really needing this. The user is now very easily able to create a template and it's not likely a common task where a user would take this path to create a template. It also adds complexity to the creation flow. We'd have to think about how they name that template and also how we surface that template being created. Add a checkbox here requires a lot :) 
I think removing it for now and stepping back to determine it's worth is the right step for this PR. WDYT?

Reply:
Makes sense to me. Especially happen about the reduction in complexity :)

Reply:
Not sure if this checkbox adds complexity. This is my take
**Small cognitive load:**
1. It has a very small footprint, short straightforward text wrapped in a checkbox
2. It is placed on a page where no action is required
3. It is not mandatory
4. Unless it is a backend complexity, the complexity that comes with this option is a "Template name" field, which can be autocompleted by the VM name itself (+"template")

**Value:**
1. I think this option does cover a scenario appealing to users: "Create a VM AND a template on the same run".
2. I also think it is not that unlikely that users might want to use this action as an afterthought after completing the Create VM wizard.

Having said that, I don't feel strongly about keeping it or removing it.

Reply:
> Not sure if this checkbox adds complexity. This is my take:
> 
> **Small cognitive load:**
> 
> 1. It has a very small footprint, short straightforward text wrapped in a checkbox
> 2. It is placed on a page where no action is required
> 3. It is not mandatory
> 4. Unless it is a backend complexity, the complexity that comes with this option is a "Template name" field, which can be autocompleted by the VM name itself (+"template")
> 

ya I don't disagree - I'm not saying we shouldn't consider it but I do think pulling it from these designs make sense until we determine it's something that will get implemented.

> **Value:**
> 
> 1. I think this option does cover a scenario appealing to users: "Create a VM AND a template on the same run".

This is interesting - it makes we wonder actually if it makes more sense to add the ability to create a VM from a template when you create the template. Basically the opposite of what we have in these designs.

> 2. I also think it is not that unlikely that users might want to use this action as an afterthought after completing the Create VM wizard.
> 
> Having said that, I don't feel strongly about keeping it or removing it.

Reply:
> This is interesting - it makes we wonder actually if it makes more sense to add the ability to create a VM from a template when you create the template. Basically the opposite of what we have in these designs.

It's just one of those things where I don't think a user would go to VM Templates expecting to create a VM.

Reply:
> It's just one of those things where I don't think a user would go to VM Templates expecting to create a VM.

I agree. I'll drop the "Create template" option for now as it becomes an independent item, but we should pursue it in the near future

Reply:
ya I really think the only way of validating any of this is from user research. It's certainly not a feature that is required for CNV but could prove in some way to be a "nice to have" that we could possibly consider.